include/psa/crypto.h

```diff
@@ -3759,6 +3759,418 @@ psa_status_t psa_generate_key(const psa_key_attributes_t *attributes,
+
+/** \def PSA_PAKE_OPERATION_INIT
+ *
+ * This macro returns a suitable initializer for an PAKE operation object of
+ * type #psa_pake_operation_t.
+ */
+#ifdef __DOXYGEN_ONLY__
+/* This is an example definition for documentation purposes.
+ * Implementations should define a suitable value in `crypto_struct.h`.
+ */
+#define PSA_PAKE_OPERATION_INIT {0}
+#endif
+
+/** Return an initial value for an PAKE operation object.
+ */
+static psa_pake_operation_t psa_pake_operation_init(void);
+
+/** Set the session information for a password-authenticated key exchange.
+ *
+ * The sequence of operations to set up a password-authenticated key exchange
+ * is as follows:
+ * -# Allocate an operation object which will be passed to all the functions
+ *    listed here.
+ * -# Initialize the operation object with one of the methods described in the
+ *    documentation for #psa_pake_operation_t, e.g.
+ *    #PSA_PAKE_OPERATION_INIT.
+ * -# Call psa_pake_setup() to specify the algorithm, the key, cipher suite,
+ *    identities and additional session information.
+ *
+ * A typical sequence of calls to perform a password-authenticated key
+ * exchange:
+ * -# Call psa_pake_get_key_share() to get the key share that needs to be sent
+ *    to the peer.
+ * -# Call psa_pake_set_key_share() to provide the key share that was received
+ *    from the peer.
+ * -# Call psa_pake_get_implicit_key() for accessing the shared secret.
+ *
+ * The exact sequence of calls to perform a password-authenticated key exchange
+ * depends on the protocol in use:
+ * -# Some protocols exchange more data than just a single key share. When using
+ *    such a protocol, call psa_pake_output() and psa_pake_input() one or more
+ *    times to exchange any further data that is needed to derive the shared
+ *    secret.
+ *
+ * Refer to the documentation of individual PAKE algorithm types (`PSA_ALG_XXX`
+ * values of type ::psa_algorithm_t such that #PSA_ALG_IS_PAKE(\c alg) is true)
+ * for more information.
+ *
+ * If an error occurs at any step after a call to psa_pake_setup(),
+ * the operation will need to be reset by a call to psa_pake_abort(). The
+ * application may call psa_pake_abort() at any time after the operation
+ * has been initialized.
+ *
+ * After a successful call to psa_pake_setup(), the application must
+ * eventually terminate the operation. The following events terminate an
+ * operation:
+ * - A call to psa_pake_abort().
+ * - A successful call to psa_pake_get_implicit_key().
+ *
+ * \param[in,out] operation     The operation object to set up. It must have
+ *                              been initialized as per the documentation for
+ *                              #psa_pake_operation_t and not yet in use.
+ * \param password              Identifier of the key holding the password or a
+ *                              value derived from the password (eg. by a
+ *                              memory-hard function).  It must remain valid
```

Comment:
> When would one use a password vs a derived value?

It would be a password if the password is used directly as it is and it would be derived when it went through a memory hard function first.

> Does it depend on the PAKE protocol?

The memory hard function applied to the password (if any) is out of scope for most standards. Some of them mention the possibility, some of them recommend, but are agnostic. This step can be done prior the execution of the PAKE.

The exception is OPAQUE where the memory hard function is integral part of the algorithm (can't be split off as a pre-processing step) and therefore (although technically possible) it doesn't make any sense to use an MHF before passing the password.